frontend/src/routes/Upload.tsx

```diff
@@ -84,7 +115,8 @@ const UploadMain: React.FC = () => {
 
     const progressHistory = useRef<ProgressHistory>([]);
 
-    useNavBlocker(() => !!uploadState.current && uploadState.current.state !== "done");
+    useNavBlocker(() => !!uploadState.current
+        && !["done", "cancelled"].includes(uploadState.current.state));
```

Comment:
This feels a little weird, but doing this with a logical OR somehow results in the following error:
<img width="480" alt="Bildschirmfoto 2023-01-11 um 16 26 44" src="https://user-images.githubusercontent.com/94838646/211847157-97d44539-2f59-4bb6-887b-f3255aee4446.png">

Reply:
I think you meant to use logical AND instead. It should black if the state is not "done" AND not "cancelled". The error is shown because in `uploadState.current.state !== "done" || RHS`, the RHS is only evaluated if the left hand side of `||` is `false` (thats called "short circuiting operators"). But if the left hand side of `||` is false that means that `state === "done"`, i.e. in that case we know for sure that state is that one particular value. Thus checking `"done" !== "cancelled"` is not useful as it's always `true`. As such the `||` expression as you wrote it would always return `true`. I hope that makes sense.

I don't mind the `includes` solution tho, so feel free to leave it.

Reply:
This conversation can be resolved, but @JulianKniephoff still wanted to review this. And I haven't had a closer look at your new commits either. Just answered this question.